litmus-portal/graphql-server/pkg/chaos-workflow/handler/handler.go

```diff
@@ -400,78 +399,151 @@ func QueryWorkflowRuns(input model.GetWorkflowRunsInput) (*model.GetWorkflowsOut
+				{"$match", bson.D{
+					{"workflow_name", bson.D{
+						{"$regex", workflowInput.Filter.WorkflowName},
+					}},
+				}},
 			}
-			result = append(result, &newChaosWorkflows)
+			pipeline = append(pipeline, matchWfNameStage)
+		}
+
+		// Filtering based on cluster name
+		if workflowInput.Filter.ClusterName != nil && *workflowInput.Filter.ClusterName != "All" && *workflowInput.Filter.ClusterName != "" {
+			matchClusterStage := bson.D{
+				{"$match", bson.D{
+					{"cluster_name", workflowInput.Filter.ClusterName},
+				}},
+			}
+			pipeline = append(pipeline, matchClusterStage)
 		}
 	}
 
-	return result, nil
-}
+	var sortStage bson.D
 
-// QueryListWorkflow returns all the workflows present in the given project
-func QueryListWorkflow(project_id string, workflowIds []*string) ([]*model.Workflow, error) {
-	var query bson.D
-	if len(workflowIds) != 0 {
-		query = bson.D{
-			{"project_id", project_id},
-			{"workflow_id", bson.M{"$in": workflowIds}},
+	switch {
+	case workflowInput.Sort != nil && workflowInput.Sort.Field == model.WorkflowSortingFieldName:
+		// Sorting based on WorkflowName time
+		if workflowInput.Sort.Descending != nil && *workflowInput.Sort.Descending {
+			sortStage = bson.D{
+				{"$sort", bson.D{
+					{"workflow_name", -1},
+				}},
+			}
+		} else {
+			sortStage = bson.D{
+				{"$sort", bson.D{
+					{"workflow_name", 1},
+				}},
+			}
 		}
-	} else {
-		query = bson.D{
-			{"project_id", project_id},
+	default:
+		// Default sorting: sorts it by LastUpdated time in descending order
+		sortStage = bson.D{
+			{"$sort", bson.D{
+				{"updated_at", -1},
+			}},
 		}
 	}
-	chaosWorkflows, err := dbOperationsWorkflow.GetWorkflows(query)
+
+	// Pagination
+	paginatedWorkflows := bson.A{
+		sortStage,
+	}
+
+	if workflowInput.Pagination != nil {
+		paginationSkipStage := bson.D{
+			{"$skip", workflowInput.Pagination.Page * workflowInput.Pagination.Limit},
+		}
+		paginationLimitStage := bson.D{
+			{"$limit", workflowInput.Pagination.Limit},
+		}
+
+		paginatedWorkflows = append(paginatedWorkflows, paginationSkipStage, paginationLimitStage)
+	}
+
+	// Add two stages where we first count the number of filtered workflow and then paginate the results
+	facetStage := bson.D{
+		{"$facet", bson.D{
+			{"total_filtered_workflows", bson.A{
+				bson.D{{"$count", "count"}},
+			}},
+			{"scheduled_workflows", paginatedWorkflows},
+		}},
+	}
+	pipeline = append(pipeline, facetStage)
+
+	// Call aggregation on pipeline
+	workflowsCursor, err := dbOperationsWorkflow.GetAggregateWorkflows(pipeline)
 
 	if err != nil {
 		return nil, err
 	}
+
 	var result []*model.Workflow
-	for _, workflow := range chaosWorkflows {
+
+	var workflows []dbSchemaWorkflow.AggregatedWorkflows
+
+	if err = workflowsCursor.All(context.Background(), &workflows); err != nil {
+		return &model.ListWorkflowsOutput{
+			TotalNoOfWorkflows: 0,
+			Workflows:          result,
+		}, nil
+	}
+
+	for _, workflow := range workflows[0].ScheduledWorkflows {
```

Comment:
Can you add an error handler for an empty workflow count?

Reply:
It is already done above @rajdas98 ^ ^
`
if err = workflowsCursor.All(context.Background(), &amp;workflows); err != nil {		
return &amp;model.ListWorkflowsOutput{
			TotalNoOfWorkflows: 0,		
	Workflows:          result,		
}, nil	}
`

Reply:
The above statement is checking the err from the `All` function
I was saying to check the length  of the workflow array for out of bound conditions
```
 if len(worflow) == 0 {
  return .....
}
```

Reply:
or

if len(workflow) > 0 {
logic ....}

Reply:
I guess the `All` method will return an error if the cursor is empty. I am not very sure about this. Will test this once again.
cc: @arkajyotiMukherjee

Reply:
@rajdas98 , I and @arkajyotiMukherjee just tested the code, and if there are no workflows found then it will return an empty array, and the for loop will not get executed. 
Code will return TotalNoOfWorkflows: 0, Workflows: [ ]